### PR TITLE
Update cargo

### LINKED ISSUE #80148: -Z build-std broken on latest nightly: could not read vendor path
### Steps to reproduce

1. `cargo new --lib example && cd example`
2. `echo "#![no_std]" > src/lib.rs`
3. `cargo +nightly check --target x86_64-unknown-linux-gnu -Z build-std=core`

(The actual target doesn't seem to matter, the `build-std` feature just requires that we pass a `--target` argument.)

The following error occurs:

> error: could not read vendor path /home/philipp/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/vendor
> 
> Caused by:
>   No such file or directory (os error 2)

### Cause

I think this issue occurs because https://github.com/rust-lang/rust/pull/80082 already landed, while the cargo-side revert in https://github.com/rust-lang/rust/pull/80081 did not. So **this issue will likely go away after the next cargo update**. I still decided to open this issue in case other people encounter the same problem.

### Meta

`rustc --version --verbose`:
```
rustc 1.47.0 (18bf6b4f0 2020-10-07)
binary: rustc
commit-hash: 18bf6b4f01a6feaf7259ba7cdae58031af1b7b39
commit-date: 2020-10-07
host: x86_64-unknown-linux-gnu
release: 1.47.0
LLVM version: 11.0
```

### PR DESCRIPTION
**Pull request by ehuss**
4 commits in d274fcf862b89264fa2c6b917b15230705257317..a3c2627fbc2f5391c65ba45ab53b81bf71fa323c
2020-12-07 23:08:44 +0000 to 2020-12-14 17:21:26 +0000
- Check if rerun-if-changed points to a directory. (rust-lang/cargo#8973)
- Implement external credential process. (RFC 2730) (rust-lang/cargo#8934)
- Revert recent build-std vendoring changes (rust-lang/cargo#8968)
- Fix the unit dependency graph with dev-dependency `links` (rust-lang/cargo#8969)

### CODE REVIEW

**Comment by ehuss** [comment]
@Mark-Simulacrum Asking for your review because this makes some build system changes.  Cargo has added some extra executables that need to be built.  Any build errors with them should be ignored.  I'm not sure if we will ultimately ship these, but I wanted to at least see if there are any build issues if we try.

File: src/bootstrap/tool.rs [hunk 1]
```diff
@@ -574,7 +574,40 @@ impl Step for Cargo {
...
+                tool: name,
+                mode: Mode::ToolRustc,
+                path,
+                is_optional_tool: true,
+                source_type: SourceType::Submodule,
+                extra_features: Vec::new(),
+            });
+        };
+
+        if cfg!(windows) {
```
Comment (Mark-Simulacrum) [inline src/bootstrap/tool.rs:594]: Hm, this seems wrong. I don't know that we actually support cross-compiling these but I would check the target here (self.target)

Comment (ehuss) [reply src/bootstrap/tool.rs:594]: Sure! Switched it to look at the target.

**Comment by ehuss** [comment]
@bors p=1 
build-std is broken on nightly, so I'd like to bump this.

**Pull request merged**